src/Compilers/Lua/Portable/Syntax/Syntax.xml

```diff
@@ -1,1742 +1,1997 @@
+            <PropertyComment>
+                <summary>The close parenthesis token.</summary>
+            </PropertyComment>
+            <Kind Name="CloseParenthesisToken" />
+        </Field>
+    </Node>
+
+    <Node Name="TableTypeSyntax" Base="TypeSyntax">
+        <TypeComment>
+            <summary>Represents a table type binding</summary>
+        </TypeComment>
+        <FactoryComment>
+            <summary>
+                Creates a new <see cref="TableTypeSyntax" /> node.
+            </summary>
+        </FactoryComment>
+        <Kind Name="TypeBinding" />
+        <Field Name="ColonToken" Type="SyntaxToken" Optional="true" Override="true">
+            <PropertyComment>
+                <summary>The colon at the start of the statement (if any).</summary>
+            </PropertyComment>
+            <Kind Name="ColonToken" />
+        </Field>
+        <Field Name="Types" Type="SeparatedSyntaxList&lt;TableElementTypeSyntax&gt;">
+            <PropertyComment>
+                <summary>A list of types.</summary>
+            </PropertyComment>
+        </Field>
+    </Node>
+
+    <Node Name="VarargTypeSyntax" Base="TypeSyntax">
+        <TypeComment>
+            <summary>Represents a vararg type binding</summary>
+        </TypeComment>
+        <FactoryComment>
+            <summary>
+                Creates a new <see cref="VarargTypeSyntax" /> node.
+            </summary>
+        </FactoryComment>
+        <Kind Name="TypeBinding" />
+    </Node>
+
+    <Node Name="TypeCastSyntax" Base="ExpressionSyntax">
+        <TypeComment>
+            <summary>Represents a type cast</summary>
+        </TypeComment>
+        <FactoryComment>
+            <summary>
+                Creates a new <see cref="TypeCastSyntax" /> node.
+            </summary>
+        </FactoryComment>
+        <Kind Name="TypeCasting" />
+        <Field Name="Left" Type="ExpressionSyntax">
+            <PropertyComment>
+                <summary>The expression on the left side of the operator.</summary>
+            </PropertyComment>
+        </Field>
+        <Field Name ="ColonColonToken" Type="SyntaxToken">
+            <PropertyComment>
+                <summary>The colon colon at the middle of the operator.</summary>
+            </PropertyComment>
+            <Kind Name="ColonColonToken" />
+        </Field>
+        <Field Name="Right" Type="TypeBindingSyntax">
+            <PropertyComment>
+                <summary>The type on the right side of the operator.</summary>
+            </PropertyComment>
+        </Field>
+    </Node>
+
+    <Node Name="TypeUnionSyntax" Base="LuaSyntaxNode">
+        <TypeComment>
+            <summary>Represents a type union</summary>
+        </TypeComment>
+        <FactoryComment>
+            <summary>
+                Creates a new <see cref="TypeUnionSyntax" /> node.
+            </summary>
+        </FactoryComment>
+        <Kind Name="TypeUnionSyntax" />
```

Comment:
I think a better name for this node is `UnionType`. Its base type is also `TypeSyntax`, not `LuaSyntaxNode`.
```suggestion
    <Node Name="UnionTypeSyntax" Base="TypeSyntax">
        <TypeComment>
            <summary>This node represents a union type.</summary>
        </TypeComment>
        <FactoryComment>
            <summary>
                Creates a new <see cref="UnionTypeSyntax" /> node.
            </summary>
        </FactoryComment>
        <Kind Name="UnionType" />
```